include/zephyr/sys/util.h

```diff
@@ -783,6 +783,34 @@ static inline void mem_xor_128(uint8_t dst[16], const uint8_t src1[16], const ui
 	mem_xor_n(dst, src1, src2, 16);
 }
 
+/**
+ * @brief Returns the number of bits set in a value
+ *
+ * @param value The value to count number of bits set of
+ * @param len The number of octets in @p value
+ */
+static inline size_t count_bits(const void *value, size_t len)
```

Comment:
I am a bit worried that `count_bits` is too generic, especially since we don't have a Zephyr prefix for these common util functions.

Reply:
The `include/zephyr/sys/math_extras.h` has somewhat similar traling/leading zero bit count helpers. Perhaps this one could be placed there? I do not have any good suggestion for the name, `count_bits` feels indeed a bit too generic.